src/containers/piercedStorage.hpp

```diff
@@ -405,7 +405,7 @@ friend class PiercedStorageIterator;
     template<typename T = value_t, typename std::enable_if<PiercedStorage<T, id_t>::has_restore()>::type * = nullptr>
     void restoreField(std::istream &stream, T &object);
 
-    void dumpField(std::ostream &stream, const std::vector<bool>::const_reference &value) const;
+    void dumpField(std::ostream &stream, std::vector<bool>::const_reference value) const;
```

Comment:
Why not just bool instead of vector::const_reference?

Reply:
A std::vector<bool> may not be a vector of bool objects. To make it space-efficient, every implementation is free to chose any memory layout it likes to store the elements of that array. Hence a reference to an element of a std::vector<bool> may not be a bool.

Reply:
Sorry, it's not really clear to me; my comment was related to std::vector<bool> element that I suppose you want to pass by value in this commit.

Reply:
The operator ```[]``` of a ```std::vector<bool>``` doesn't return a ```bool``` but a ```std::vector<bool>::reference```. 

Just to be sure, I will do some tests to check if, after the change, the above template specialization is still used when dumping a PiercedStorage of bools.